packages/driver/test/client.test.ts

```diff
@@ -2080,6 +2083,43 @@ test("pretty error message", async () => {
   );
 });
 
+test("warnings handler", async () => {
+  if (getEdgeDBVersion().major < 6) return;
+
+  let client = getClient();
+
+  try {
+    let warnings: EdgeDBError[] | null = null;
+    client = client.withWarningHandler((_warnings) => (warnings = _warnings));
+
+    await expect(client.query("select _warn_on_call();")).resolves.toEqual([0]);
```

Comment:
This is some kind of built-in?

Reply:
Yes, Sully 'documented' it here: https://github.com/edgedb/edgedb/pull/7823. I think it only exists in test mode/dev mode though.